backend/finance/models.py

```diff
@@ -394,3 +400,161 @@ class ReceiptDownloadToken(models.Model):
+                "self_county": invoice_instance.self_county,
+                "self_country": invoice_instance.self_country,
+            },
+            "financial_details": {
+                "sort_code": invoice_instance.sort_code,
+                "account_holder_name": invoice_instance.account_holder_name,
+                "account_number": invoice_instance.account_number,
+                "reference": invoice_instance.reference,
+                "invoice_number": invoice_instance.invoice_number,
+                "vat_number": invoice_instance.vat_number,
+                "currency": invoice_instance.currency,
+                "discount_amount": decimal_to_float(invoice_instance.discount_amount),
+                "discount_percentage": decimal_to_float(invoice_instance.discount_percentage),
+            },
+            "dates": {
+                "date_created": (
+                    invoice_instance.date_created.isoformat()
+                    if isinstance(invoice_instance.date_created, date)
+                    else invoice_instance.date_created
+                ),
+                "date_issued": (
+                    invoice_instance.date_issued.isoformat()
+                    if isinstance(invoice_instance.date_issued, date)
+                    else invoice_instance.date_issued
+                ),
+                "date_due": (
+                    invoice_instance.date_due.isoformat() if isinstance(invoice_instance.date_due, date) else invoice_instance.date_due
+                ),
+            },
+            "status": invoice_instance.status,
+            "items": [
+                {
+                    "name": item.name,
+                    "description": item.description,
+                    "is_service": item.is_service,
+                    "hours": decimal_to_float(item.hours) if item.is_service else None,
+                    "price_per_hour": decimal_to_float(item.price_per_hour) if item.is_service else None,
+                    "price": decimal_to_float(item.price),
+                }
+                for item in invoice_instance.items.all()
+            ],
+            "totals": {
+                "subtotal": decimal_to_float(invoice_instance.get_subtotal()),
+                "tax": decimal_to_float(invoice_instance.get_tax()),
+                "total": decimal_to_float(invoice_instance.get_total_price()),
+            },
+            "logo": invoice_instance.logo.url if invoice_instance.logo else None,
+            "notes": invoice_instance.notes,
+        }
+
+        history_entry = InvoiceHistory(invoice=invoice_instance, version=invoice_instance.history.count() + 1, changes=history_data)
+        history_entry.save()
+
+    @staticmethod
+    def restore_version(invoice_instance, version, save=True):
+        # Load version from history
+        history = InvoiceHistory.objects.filter(invoice=invoice_instance, version=version).first()
+        if history:
```

Comment:
Could this be if not history return, so everything can be unindented 1 more level?

Reply:
Yeah as long as there's no else or extra logic at the bottom (I'm on mobile atm so I can't see) then yeah you can return early with if not

Reply:
Ok